packages/ruleset/src/functions/integer-attributes.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * Copyright 2017 - 2024 IBM Corporation.
+ * SPDX-License-Identifier: Apache2.0
+ */
+
+const {
+  isIntegerSchema,
+  validateNestedSchemas,
+} = require('@ibm-cloud/openapi-ruleset-utilities');
+
+const { getCompositeSchemaAttribute, LoggerFactory } = require('../utils');
+
+let ruleId;
+let logger;
+module.exports = function (schema, _opts, context) {
+  if (!logger) {
+    ruleId = context.rule.name;
+    logger = LoggerFactory.getInstance().getLogger(ruleId);
+  }
+  return validateNestedSchemas(
+    schema,
+    context.path,
+    integerBoundaryErrors,
+    true,
+    false
+  );
+};
+
+// These are the valid format values that can be used with integer schemas.
+const intFormats = ['int32', 'int64'];
+
+/**
+ * This function performs various checks on an integer schema property to make sure it
+ * contains the "minimum", "maximum" attributes.
+ * @param {*} schema the schema to check
+ * @param {*} path the array of path segments indicating the "location" of the pathItem within the API definition
+ * @returns an array containing the violations found or [] if no violations
+ */
+function integerBoundaryErrors(schema, path) {
+  logger.debug(
+    `${ruleId}: checking attributes of schema at location: ${path.join('.')}`
+  );
+
+  const errors = [];
+
+  if (isIntegerSchema(schema)) {
```

Comment:
I know this is picky but I know in the past we've taken this pattern to avoid initializing objects for too many iterations where it isn't needed. (We don't need to make this change, just wanted to point it out.)

```suggestion
  if (isIntegerSchema(schema)) {
    const errors = [];
```

Reply:
fixed